lms/views/api/canvas/pages.py

```diff
@@ -29,3 +40,41 @@ def list_pages(self):
             }
             for page in self.canvas.api.pages.list(course_id)
         ]
+
+    @view_config(request_method="GET", route_name="canvas_api.pages.via_url")
+    def via_url(self):
+        application_instance = self.request.lti_user.application_instance
+        assignment = self.request.find_service(name="assignment").get_assignment(
+            application_instance.tool_consumer_instance_guid,
+            self.request.matchdict["resource_link_id"],
+        )
+
+        # We build a token to authorize the view that fetches the actual
+        # canvas pages content as the user making this request.
+        auth_token = BearerTokenSchema(self.request).authorization_param(
+            self.request.lti_user
+        )
+        return {
+            "via_url": helpers.via_url(
+                self.request,
+                self.request.route_url(
+                    "canvas_api.pages.proxy",
+                    _query={
+                        "document_url": assignment.document_url,
+                        "authorization": auth_token,
+                    },
+                ),
+            )
+        }
+
+    @view_config(request_method="GET", route_name="canvas_api.pages.proxy")
+    def proxy(self):
```

Comment:
This is mostly a placeholder to allow end-to-end testing right now.

It's missing:

- Some headers for caching
- Use a jinaj2 template
- Inject the page title on that template
- Inject a canonical URL on that page
- Inject CSS style on the template.

Added these as drafts to the board for now: https://github.com/orgs/hypothesis/projects/126/views/1